vkcs/resource_vkcs_publicdns_record.go

```diff
@@ -0,0 +1,411 @@
+	recordTypeAArgs      = [1]string{"ip_v4"}
+	recordTypeAAAAArgs   = [1]string{"ip_v6"}
+	recordTypeCNAMEArgs  = [1]string{"content"}
+	recordTypeMXArgs     = [2]string{"priority", "content"}
+	recordTypeNSArgs     = [1]string{"content"}
+	recordTypeSRVArgs    = [6]string{"service", "proto", "priority", "weight", "host", "port"}
+	recordTypeTXTArgs    = [1]string{"content"}
+)
+
+func resourcePublicDNSRecord() *schema.Resource {
+	return &schema.Resource{
+		CustomizeDiff: resourcePublicDNSRecordCustomizeDiff,
+
+		CreateContext: resourcePublicDNSRecordCreate,
+		ReadContext:   resourcePublicDNSRecordRead,
+		UpdateContext: resourcePublicDNSRecordUpdate,
+		DeleteContext: resourcePublicDNSRecordDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: func(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				zoneID, recordType, recordID, err := publicDNSRecordParseID(d.Id())
+				if err != nil {
+					return nil, err
+				}
+
+				recordType = strings.ToUpper(recordType)
+				d.Set("zone_id", zoneID)
+				d.Set("type", recordType)
+				d.SetId(recordID)
+
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(recordCreateTimeout),
+			Delete: schema.DefaultTimeout(recordDeleteTimeout),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The region in which to obtain the V2 Public DNS client. If omitted, the `region` argument of the provider is used. Changing this creates a new record.",
+			},
+
+			"zone_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the zone to attach the record to.",
+			},
+
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					recordTypeA,
+					recordTypeAAAA,
+					recordTypeCNAME,
+					recordTypeMX,
+					recordTypeNS,
+					recordTypeSRV,
+					recordTypeTXT,
+				}, false),
+				Description: "The type of the record. Must be one of following: \"A\", \"AAAA\", \"CNAME\", \"MX\", \"NS\", \"SRV\", \"TXT\".",
+			},
+
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The name of the record.",
+			},
+
+			"ip_v4": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The IPv4 of the record.",
+			},
+
+			"ip_v6": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The IPv6 of the record.",
+			},
+
+			"content": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The content of the record.",
```

Comment:
Should not we say smth like "the content of CNAME, MX, NS, TXT record"? The point is to specify somewhere what arguments are available for certain record types